provider/src/androidMain/kotlin/at/asitplus/crypto/provider/os/AndroidKeyStoreProvider.kt

```diff
@@ -0,0 +1,402 @@
+package at.asitplus.crypto.provider.os
+
+import android.annotation.SuppressLint
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyInfo
+import android.security.keystore.KeyProperties
+import android.security.keystore.UserNotAuthenticatedException
+import androidx.biometric.BiometricPrompt
+import androidx.biometric.BiometricPrompt.AuthenticationResult
+import androidx.biometric.BiometricPrompt.CryptoObject
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.crypto.datatypes.CryptoPublicKey
+import at.asitplus.crypto.datatypes.CryptoSignature
+import at.asitplus.crypto.datatypes.Digest
+import at.asitplus.crypto.datatypes.RSAPadding
+import at.asitplus.crypto.datatypes.SignatureAlgorithm
+import at.asitplus.crypto.datatypes.getJCASignatureInstance
+import at.asitplus.crypto.datatypes.jcaName
+import at.asitplus.crypto.datatypes.parseFromJca
+import at.asitplus.crypto.datatypes.pki.CertificateChain
+import at.asitplus.crypto.datatypes.pki.X509Certificate
+import at.asitplus.crypto.datatypes.pki.leaf
+import at.asitplus.crypto.provider.UnlockFailed
+import at.asitplus.crypto.provider.UnsupportedCryptoException
+import at.asitplus.crypto.provider.dsl.DSL
+import at.asitplus.crypto.provider.dsl.DSLConfigureFn
+import at.asitplus.crypto.provider.sign.SignatureInput
+import com.ionspin.kotlin.bignum.integer.base63.toJavaBigInteger
+import at.asitplus.crypto.provider.sign.Signer as SignerI
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import java.security.KeyFactory
+import java.security.KeyPairGenerator
+import java.security.KeyStore
+import java.security.PrivateKey
+import java.security.Signature
+import java.security.spec.ECGenParameterSpec
+import java.security.spec.RSAKeyGenParameterSpec
+import java.time.Instant
+import java.util.Date
+import javax.security.auth.x500.X500Principal
+import kotlin.enums.enumEntries
+import java.security.Signature as JCASignatureObject
+
+internal sealed interface FragmentContext {
+    @JvmInline value class OfActivity(val activity: FragmentActivity): FragmentContext
+    @JvmInline value class OfFragment(val fragment: Fragment): FragmentContext
+}
+
+
+class AndroidTPMConfiguration internal constructor(): PlatformSigningKeyConfiguration.TPMConfiguration() {
```

Comment:
`TMP` -> `SE`. this applies to the whole codebase

Reply:
Is "secure enclave" a generic term that android devs would recognize? I primarily associate it with Apple-world.

Can we maybe find a more generic term... `hardware-backed` or something?

Reply:
SE as in "secue element" hardware-backed is fine too. The code should read nicely in the end

Reply:
renames are done, check when you have time